run_test.py

```diff
@@ -0,0 +1,160 @@
+                '--ez', 'performancetest', 'true'
+            ])
+        else:
+            self._shell([
+                'am', 'start-activity',
+                '-t', 'text/html', '-d', page,
+                '-a', 'android.intent.action.VIEW',
+                '{}/{}'.format(self.package_name, self.applink_component),
+                '--ez', 'performancetest', 'true'
+            ])
+
+    def keyevent(self, event='HOME'):
+        self._shell(['input', 'keyevent', event])
+
+
+class FenixAdbConnection(AdbConnection):
+    home_component = 'org.mozilla.fenix.HomeActivity'
+    applink_component = 'org.mozilla.fenix.IntentReceiverActivity'
+
+
+class FennecAdbConnection(AdbConnection):
+    home_component = '.App'
+    applink_component = 'org.mozilla.gecko.LauncherActivity'
+
+
+def connect_adb(package_name):
+    if ( re.match(r'org\.mozilla\.fenix\.?', package_name) or
+         package_name == 'org.mozilla.fenix_aurora' ):
+        return FenixAdbConnection(package_name)
+    else:
+        return FennecAdbConnection(package_name)
+
+
+def report_error(error, filename):
+    if filename:
+        with open(filename, 'w') as f:
+            print(str(error), file=f)
+    else:
+        print('error: {}'.format(error), file=sys.stderr)
+    exit(1)
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument('name', help='package name')
+    parser.add_argument('-I', '--install', metavar='PATH',
+                        help='APK to install')
+    parser.add_argument('-F', '--file', metavar='PATH',
+                        help='output file for logcat results')
+
+    exec_p = parser.add_argument_group('execution arguments')
+    exec_p.add_argument('-n', '--num', type=int, default=10, metavar='N',
+                        help='number of test runs (default: %(default)s)')
+    exec_p.add_argument('-W', '--warmup', type=int, default=1, metavar='N',
+                        help=('number of test runs to warm up with before ' +
+                              'recording (default: %(default)s)'))
+    exec_p.add_argument('-p', '--page', metavar='URL',
+                        help='page to load for app-link')
+
+    time_p = parser.add_argument_group('timing arguments')
+    time_p.add_argument('--between-time', type=float, default=1.0, metavar='T',
+                        help=('time to wait between runs, in seconds ' +
+                              '(default: %(default)s)'))
+    time_p.add_argument('--start-time', type=float, default=5.0, metavar='T',
+                        help=('time to wait after starting process, in ' +
+                              'seconds (default: %(default)s)'))
+    time_p.add_argument('--finish-time', type=float, default=5.0, metavar='T',
+                        help=('time to wait until killing process, in ' +
+                              'seconds (default: %(default)s)'))
+
+    args = parser.parse_args()
+
+    adb = connect_adb(args.name)
```

Comment:
I'd probably want to write this all as a function so it's re-usable and testable.